TestNameValidation is flaky

```
[2019-10-07T12:47:32.142Z]         --- PASS: TestNameValidation/longestPossibleName/Creating_APM_Server_should_succeed (0.08s)

[2019-10-07T12:47:32.142Z]         --- PASS: TestNameValidation/longestPossibleName/APM_Server_should_be_created (0.07s)

[2019-10-07T12:47:32.142Z]         --- FAIL: TestNameValidation/longestPossibleName/ES_certificate_authority_should_be_set_and_deployed (300.00s)

[2019-10-07T12:47:32.142Z]             utils.go:80: 

[2019-10-07T12:47:32.142Z]                 	Error Trace:	utils.go:80

[2019-10-07T12:47:32.142Z]                 	Error:      	Received unexpected error:

[2019-10-07T12:47:32.142Z]                 	            	secrets "es-naming-54hr-xxxxxxxxxxxxxxxxxxxxx-es-transport-ca-internal" not found

[2019-10-07T12:47:32.142Z]                 	Test:       	TestNameValidation/longestPossibleName/ES_certificate_authority_should_be_set_and_deployed
```

```
NAME                                                                              HEALTH   NODES   VERSION   PHASE     AGE
elasticsearch.elasticsearch.k8s.elastic.co/es-naming-jq59-xxxxxxxxxxxxxxxxxxxxx                    7.2.0     Invalid   7m55s
```

```
  - name: yyyyyyyyyyyyyyyyyyyyyyy
    podTemplate:
      metadata:
        creationTimestamp: null
      spec:
        containers: null
```

This is because there is no `count` in the spec

--- a/pkg/controller/elasticsearch/name/name.go
+++ b/pkg/controller/elasticsearch/name/name.go
@@ -29,6 +29,8 @@ const (
 	defaultPodDisruptionBudget        = "default"
 	scriptsConfigMapSuffix            = "scripts"
 	transportCertificatesSecretSuffix = "transport-certificates"
+
+	controllerRevisionHashLen = 10
 )
 
 var (
@@ -70,11 +72,11 @@ func Validate(es v1beta1.Elasticsearch) error {
 
 		// length of the ordinal suffix that will be added to the pods of this sset (dash + ordinal)
 		podOrdinalSuffixLen := len(strconv.FormatInt(int64(nodeSet.Count), 10)) + 1
-		// there should be enough space for the ordinal suffix
-		if validation.DNS1123SubdomainMaxLength-len(ssetName) < podOrdinalSuffixLen {
+		// there should be enough space for the ordinal suffix and the controller revision hash
+		if validation.LabelValueMaxLength-len(ssetName) < podOrdinalSuffixLen+controllerRevisionHashLen {
 			return fmt.Errorf("generated StatefulSet name '%s' exceeds allowed length of %d",
 				ssetName,
-				validation.DNS1123SubdomainMaxLength-podOrdinalSuffixLen)
+				validation.LabelValueMaxLength-podOrdinalSuffixLen-controllerRevisionHashLen)
 		}
 	}
 

--- a/test/e2e/es/naming_test.go
+++ b/test/e2e/es/naming_test.go
@@ -6,6 +6,8 @@ package es
 
 import (
 	"fmt"
+	"hash/fnv"
+	"strconv"
 	"strings"
 	"testing"
 
@@ -31,11 +33,19 @@ func TestNameValidation(t *testing.T) {
 
 func testLongestPossibleName(t *testing.T) {
 	maxESNameLen := name.MaxResourceNameLength
-	maxNodeSpecNameLen := validation.LabelValueMaxLength - maxESNameLen - 4
 	randSuffix := rand.String(4)
 
 	esNamePrefix := strings.Join([]string{"es-naming", randSuffix}, "-")
 	esName := strings.Join([]string{esNamePrefix, strings.Repeat("x", maxESNameLen-len(esNamePrefix)-1)}, "-")
+
+	// StatefulSet name would look like <esName>-es-<nodeSpecName>
+	// Pods created by the StatefulSet will have the ordinal appended to the name and a controller revision hash
+	// label created by appending a revision hash to the pod name.
+	revisionHash := fnv.New32()
+	revisionHash.Write([]byte("some random data"))
+	fullRevisionHash := rand.SafeEncodeString(strconv.FormatInt(int64(revisionHash.Sum32()), 10))
+
+	maxNodeSpecNameLen := validation.LabelValueMaxLength - len(esName) - len("-es-") - len("-0") - len(fmt.Sprintf("-%s", fullRevisionHash))
 	nodeSpecName := strings.Repeat("y", maxNodeSpecNameLen)
 	esBuilder := elasticsearch.NewBuilderWithoutSuffix(esName).
 		WithESMasterDataNodes(3, elasticsearch.DefaultResources).
@@ -43,7 +53,8 @@ func testLongestPossibleName(t *testing.T) {
 		WithVersion(test.Ctx().ElasticStackVersion).
 		WithRestrictedSecurityContext().
 		WithNodeSpec(estype.NodeSet{
-			Name: nodeSpecName,
+			Name:  nodeSpecName,
+			Count: 1,
 		})
 
 	kbNamePrefix := strings.Join([]string{esNamePrefix, "kb"}, "-")
@@ -79,7 +90,8 @@ func testRejectionOfLongName(t *testing.T) {
 		WithVersion(test.Ctx().ElasticStackVersion).
 		WithRestrictedSecurityContext().
 		WithNodeSpec(estype.NodeSet{
-			Name: "default",
+			Name:  "default",
+			Count: 1,
 		})
 
 	objectCreated := false

--- a/test/e2e/es/naming_test.go
+++ b/test/e2e/es/naming_test.go
@@ -42,7 +42,7 @@ func testLongestPossibleName(t *testing.T) {
 	// Pods created by the StatefulSet will have the ordinal appended to the name and a controller revision hash
 	// label created by appending a revision hash to the pod name.
 	revisionHash := fnv.New32()
-	revisionHash.Write([]byte("some random data"))
+	_, _ = revisionHash.Write([]byte("some random data"))
 	fullRevisionHash := rand.SafeEncodeString(strconv.FormatInt(int64(revisionHash.Sum32()), 10))
 
 	maxNodeSpecNameLen := validation.LabelValueMaxLength - len(esName) - len("-es-") - len("-0") - len(fmt.Sprintf("-%s", fullRevisionHash))

